### PR TITLE
Feature/axis labels

### PR DESCRIPTION
**Pull request by robwebdev**

### CODE REVIEW

File: src/Data/Widget/Adapters/CellPosition.elm [hunk 1]
```diff
@@ -9,6 +9,11 @@ type CellPosition
     = CellPosition ( Int, Int )
 
 
+defaultPosition : CellPosition
+defaultPosition =
+    CellPosition ( 1, 1 )
```
Comment (msp) [inline src/Data/Widget/Adapters/CellPosition.elm:14]: +1

File: src/Data/Widget/Adapters/CellRange.elm [hunk 2]
```diff
@@ -16,11 +27,11 @@ type alias CellRange =
     }
 
 
-emptyRange : CellRange
-emptyRange =
+defaultRange : CellRange
+defaultRange =
```
Comment (msp) [inline src/Data/Widget/Adapters/CellRange.elm:31]: +1 +1

File: src/Data/Widget/Adapters/LineAndBarAdapter.elm [hunk 3]
```diff
@@ -80,13 +82,19 @@ extractChartConfig combinedConfig rowsKey seriesLabelKey =
             cleansedConfig
 
 
-adapt : AdapterConfig.Config -> Table.Data -> ( Chart.Data, Chart.Data )
+adapt : AdapterConfig.Config -> Table.Data -> ( Chart.Data, Chart.Data, Maybe String, Maybe String )
```
Comment (msp) [inline src/Data/Widget/Adapters/LineAndBarAdapter.elm]: We should encapsulate these in Chart.Data?

File: src/Data/Widget/Chart.elm [hunk 4]
```diff
@@ -11,4 +11,6 @@ type alias Data =
     , maxValue : Float
     , xLabels : List String
     , seriesLabels : Maybe (List String)
+    , xAxisLabel : Maybe String
```
Comment (msp) [inline src/Data/Widget/Chart.elm:14]: Oh, they are here already?

File: src/Views/Widget/Renderers/BarChart.elm [hunk 5]
```diff
@@ -138,14 +141,14 @@ column height index totalRows colour xScaleBand maxValue ( header, value ) =
             ]
 
 
-view : Int -> Int -> List (List ( Cell, Cell )) -> Float -> Maybe (List String) -> Svg msg
-view w h data maxValue seriesLabels =
+view : Int -> Int -> Chart.Data -> Svg msg
```
Comment (msp) [inline src/Views/Widget/Renderers/BarChart.elm:139]: Thank you!

File: src/Views/Widget/Renderers/LineChart.elm [hunk 6]
```diff
@@ -75,15 +77,15 @@ getLineColour index =
         |> Color.Convert.colorToHex
 
 
-view : Int -> Int -> List (List ( Cell, Cell )) -> Float -> Maybe (List String) -> Svg msg
-view w h data maxValue seriesLabels =
+view : Int -> Int -> Chart.Data -> Svg msg
```
Comment (msp) [inline src/Views/Widget/Renderers/LineChart.elm:75]: +1

**Review by msp — COMMENTED** [review]
Looking good.

Maybe I'm missing something but not sure why we need to explicitly pass around the labels if they're encapsulated in Chart.Data ?

**Comment by msp** [comment]
Just figured out the missing piece! Combo charts likely need 2 y-axis labels, one for each side. This is prob the most important usage actually as there are 2 sets of numbers which look confusing without labels.

**Comment by robwebdev** [comment]
Good catch about the missing combo chart axis! I'll address that. 

Now I can see that the y-axis labels for the combo chart can live inside each respective`Chart.Data` now that each type will have it's own label. But who owns the x-axis label? It's neither the line or the bar `Chart.Data` which is why I had `LineAndBarAdpater.adapt` returning the labels outside of the data.

**Comment by robwebdev** [comment]
Actually seeing as the x-axis is being rendered from the line chart it makes sense that the label lives in that `Chart.Data`. I'm with you now I think 👍

**Comment by robwebdev** [comment]
Rebased to bring in the CI config. Ready for another review @msp

**Comment by msp** [comment]
Great, thanks. Code & functionality now look good. 

We're gonna have to do something to format the axis tickets though as the labels are very close. Would you be able to take a look? I can merge this one if you prefer to start a fresh branch..

That'll free me up tmrw to look at the Forecast feature.

screenshot 2018-02-19 12 18 01

**Comment by robwebdev** [comment]
screen shot 2018-02-20 at 15 38 35

**Review by msp — APPROVED** [review]

**Pull request merged**